Update to Python 3

tfrisk-old opened the pull request:
I wanted to test drive your script but had problems running it with Python 3.4. I managed to get the base script working with these changes but there might be some issues hiding.

Testing against live IOT devices is not done because I don't have suitable targets available at this moment.

tfrisk committed:
Update to Python 3

fernviridian commented:
I just got a chance to check out the new python code and I'm running into type issues with urllib. Seems to be a common problem with lots of posts on stackoverflow. Maybe the requests library would work, since we are just using simple get requests.  

```
Loading 'modules.sucasa' as a 'simple' idIOTic execute module...
Loading 'modules.wemo' as a 'simple' idIOTic execute module...


            ...oooOOO The Internet of Things is idIOTic OOOooo...


idIOTic.>> sucasa IPADDR
[+] grabbing /etc/passwd...

b'root:$1$IHRYUhFB$hQigs1sQUSlxNdLAuVD4N.:0:0:root:/root:/bin/ash\nnobody:*:65534:65534:nobody:/var:/bin/false\ndaemon:*:65534:65534:daemon:/var:/bin/false\n'
Traceback (most recent call last):
  File "idIOTic.py", line 153, in <module>
    ui_classref("idIOTic", """""").cmdloop()
  File "/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/cmd.py", line 138, in cmdloop
    stop = self.onecmd(line)
  File "/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/cmd.py", line 217, in onecmd
    return func(arg)
  File "/Users/ben/repos/idIOTic/modules/sucasa.py", line 24, in execute
    password = str(urlopen(url).read().split("-")[1].split(" to the")[0])
TypeError: Type str doesn't support the buffer API
```

tfrisk-old commented:
I checked the code with Pythons own <code>2to3</code> script (https://docs.python.org/2/library/2to3.html , http://getpython3.com/diveintopython3/porting-code-to-python-3-with-2to3.html) and it suggested the following change to <code>urlopen</code> commands: 

<pre>
-        password = str(urllib2.urlopen(url).read().split("-")[1].split(" to the")[0])
+        password = str(urllib.request.urlopen(url).read().split("-")[1].split(" to the")[0])
</pre>

Not sure if it fixes the type errors but worth a try.

I think I cannot fix this any further because I don't have the means to test the actual download.

fernviridian commented:
Hi tfrisk, I've been swamped with work lately. Sorry I didn't get to this sooner. I've tested the suggested updates to Python3 and your code changes. Everything seems to be working fine. Thank you for the work! I will be pushing an update to the repo here really soon to fix the type errors.
